### PR TITLE
js3 week3 hw added

### PR DESCRIPTION
**Pull request by ccshah298**

### CODE REVIEW

File: javascript/javascript3/week3/class.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  constructor() {
+    this.products = [];
+  }
+  addProduct(product) {
+    // Implement functionality here
+    this.products.push(product);
+  }
+  removeProduct(product) {
+    // Implement functionality here
+    this.products = this.products.filter((cart) => cart.name !== product.name);
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/class.js:22]: good job using filter. Clean and most people didn't do that.

File: javascript/javascript3/week3/class.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        <div> Product_Name: ${product.name}</div>
+        <div> Price: ${product.price}</div>`;
+
+      shopping.appendChild(productItem);
+    });
+    const total = document.createElement("li");
+    total.innerHTML = `User Name :${this.getUser()} , Total Price ${this.getTotal()}`;
+    shopping.appendChild(total);
+  }
+  getUser() {
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/class.js:56]: use async / await. Also pass the user name as a parameter.

File: javascript/javascript3/week3/class.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+// Shopping cart using Classes
+class Product {
+  constructor(name, price) {
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/class.js:3]: what currency units is this price specified in? The variable naming should reflect that.

File: javascript/javascript3/week3/promise.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+//Getting into promises
+//Select 3 classmates github username that you want to show repositories for.
+const username1 = "sandhana14";
+const username2 = "ccshah298";
+const username3 = "JyothiNandyala";
+//Fetch all the 3 classmates repositories
+const promise1 = fetch(
+  `https://api.github.com/search/repositories?q=user:${username1}`
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/promise.js:8]: use a separate const for `https://api.github.com/search/repositories?q=user:` Or extract it into a function that takes a username as a parameter.

File: javascript/javascript3/week3/promise.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  userName.innerHTML = data.innerHTML = data.items[0].owner.login;
+  data.items.forEach((item) => {
+    const ul = document.createElement("ul");
+    ul.innerHTML = `<li>Name: ${item.name}</li>
+               <li>Url: <a href="${item.url}">${item.url}</a></li>
+               <li>Owner: ${item.owner.login}</li>`;
+    mainDiv.appendChild(ul);
+  });
+}
+Promise.all([promise1, promise2, promise3])
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/promise.js:30]: good use of Promise.all

File: javascript/javascript3/week3/promise.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  .then((responses) => Promise.all(responses.map((r) => r.json())))
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/promise.js:31]: i think you can do this part without `Promise.all`

**Review by aleksd-dfds — APPROVED** [review]
Good job, the code seems to work, but a few style and readability improvements can be made. Especially async / await

File: javascript/javascript3/week3/promise.js (hunk 6 above)
Comment (ccshah298) [reply javascript/javascript3/week3/promise.js:31]: Thanks Aleks for reviewing my HW..

**Pull request merged**